File: python/baiduads-sdk/test/marketingfile/marketing_file_service_test.py
"""
for test MarketingFileService
"""
import time
import unittest

from baiduads.common.model import ApiRequestHeader
from baiduads_manual.marketingfile.api.marketing_file_service import MarketingFileService
from baiduads_manual.marketingfile.model.upload_file_params import UploadFileParams
from baiduads_manual.marketingfile.model.upload_file_request import UploadFileRequest
from baiduads_manual.marketingfile.model.upload_file_request_wrapper import UploadFileRequestWrapper


class TestMarketingFileService(unittest.TestCase):

    def setUp(self):
        self.api = MarketingFileService()

    def tearDown(self):
        pass

    def test_upload_file_simple(self):
        file = "/tmp/a.txt"
        # 请填写自己的信息
        header = ApiRequestHeader(
            token="xxxxx",
            username="xxxxx",
            password="xxxxx",
            target="xxxxx",
            _spec_property_naming=True
        )
        body = UploadFileRequest(
            file=file,
            params=UploadFileParams(fileName="test" + str(time.time()) + ".txt",
                                    fileType="txt",
                                    storeType="temp",
                                    bizFlag="audience")
        )

        upload_file_request_wrapper = UploadFileRequestWrapper(
            header=header,
            body=body
        )
        res = self.api.upload_file(upload_file_request_wrapper)
        assert res.header.status == 0
        print(res)

    def test_upload_file_simple_with_timeout(self):
        file = "/tmp/a.txt"
        # 请填写自己的信息
        header = ApiRequestHeader(
            token="xxxxx",
            username="xxxxx",
            password="xxxxx",
            target="xxxxx",
            _spec_property_naming=True
        )
        body = UploadFileRequest(
            file=file,
            params=UploadFileParams(fileName="test" + str(time.time()) + ".txt",
                                    fileType="txt",
                                    storeType="temp",
                                    bizFlag="audience"),
            timeout=500
        )

        upload_file_request_wrapper = UploadFileRequestWrapper(
            header=header,
            body=body
        )
        res = self.api.upload_file(upload_file_request_wrapper)
        print(res)
        assert res.header.status == 0

    def test_upload_file_multipart(self):
        file = "/tmp/b.txt"
        # 请填写自己的信息
        header = ApiRequestHeader(
            token="xxxxx",
            username="xxxxx",
            password="xxxxx",
            target="xxxxx",
            _spec_property_naming=True
        )
        body = UploadFileRequest(
            file=file,
            params=UploadFileParams(fileName="test" + str(time.time()) + ".txt",
                                    fileType="txt",
                                    storeType="temp",
                                    bizFlag="audience")
        )

        upload_file_request_wrapper = UploadFileRequestWrapper(
            header=header,
            body=body
        )
        res = self.api.upload_file(upload_file_request_wrapper)
        print(res)
        assert res.header.status == 0

    def test_upload_file_multipart_with_timeout(self):
        file = "/tmp/b.txt"
        # 请填写自己的信息
        header = ApiRequestHeader(
            token="xxxxx",
            username="xxxxx",
            password="xxxxx",
            target="xxxxx",
            _spec_property_naming=True
        )
        body = UploadFileRequest(
            file=file,
            params=UploadFileParams(fileName="test" + str(time.time()) + ".txt",
                                    fileType="txt",
                                    storeType="temp",
                                    bizFlag="audience"),
            timeout=10000
        )

        upload_file_request_wrapper = UploadFileRequestWrapper(
            header=header,
            body=body
        )
        res = self.api.upload_file(upload_file_request_wrapper)
        print(res)
        assert res.header.status == 0


if __name__ == '__main__':
    unittest.main()

File: java/baiduads-sdk/src/test/com/baidu/dev2/api/sdk/marketingfile/MarketingFileServiceTest.java
package com.baidu.dev2.api.sdk.marketingfile;

import com.baidu.dev2.api.sdk.common.ApiRequestHeader;
import com.baidu.dev2.api.sdk.manual.marketingfile.api.MarketingFileService;
import com.baidu.dev2.api.sdk.manual.marketingfile.model.UploadFileParams;
import com.baidu.dev2.api.sdk.manual.marketingfile.model.UploadFileRequest;
import com.baidu.dev2.api.sdk.manual.marketingfile.model.UploadFileRequestWrapper;
import com.baidu.dev2.api.sdk.manual.marketingfile.model.UploadFileResponseWrapper;
import com.baidu.dev2.api.sdk.manual.upload.model.MultiUploadOptions;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import java.io.File;

public class MarketingFileServiceTest {

    private MarketingFileService service;
    private ApiRequestHeader header;

    @Before
    public void before() {
        service = new MarketingFileService();
        header = new ApiRequestHeader();
        // 请填写自己的信息
        header.setAccessToken("xxxxx"); // 替换为授权生成的accessToken
        header.setUserName("xxxxx"); // 替换为自己的用户名
    }

    /***
     * 测试简单文件上传
     * @throws Exception
     */
    @Test
    public void testUploadFileSimple() throws Exception {
        UploadFileRequestWrapper uploadFileRequestWrapper = new UploadFileRequestWrapper();
        uploadFileRequestWrapper.setHeader(header);

        File file = new File("/tmp/a.txt");

        UploadFileRequest uploadFileRequest = new UploadFileRequest();
        uploadFileRequest.setFile(file);

        UploadFileParams uploadFileParams = new UploadFileParams();
        uploadFileParams.setFileName("test" + System.currentTimeMillis() + ".txt");
        uploadFileParams.setFileType("txt");
        uploadFileParams.setStoreType("temp");
        uploadFileParams.setBizFlag("audience");
        uploadFileRequest.setParams(uploadFileParams);

        uploadFileRequestWrapper.setBody(uploadFileRequest);

        UploadFileResponseWrapper uploadFileResponseWrapper = service.uploadFile(uploadFileRequestWrapper);
        System.out.println(uploadFileResponseWrapper);
        Assert.assertTrue(uploadFileResponseWrapper.getHeader().getStatus() == 0);
    }

    /**
     * 测试大文件的分片上传
     * @throws Exception
     */
    @Test
    public void testUploadFileMulti() throws Exception {
        UploadFileRequestWrapper uploadFileRequestWrapper = new UploadFileRequestWrapper();
        uploadFileRequestWrapper.setHeader(header);

        File file = new File("/tmp/b.txt");

        UploadFileRequest uploadFileRequest = new UploadFileRequest();
        uploadFileRequest.setFile(file);

        UploadFileParams uploadFileParams = new UploadFileParams();
        uploadFileParams.setFileName("test" + System.currentTimeMillis() + ".txt");
        uploadFileParams.setFileType("txt");
        uploadFileParams.setStoreType("temp");
        uploadFileParams.setBizFlag("audience");
        uploadFileRequest.setParams(uploadFileParams);

        uploadFileRequestWrapper.setBody(uploadFileRequest);

        UploadFileResponseWrapper uploadFileResponseWrapper = service.uploadFile(uploadFileRequestWrapper);
        System.out.println(uploadFileResponseWrapper);
        Assert.assertTrue(uploadFileResponseWrapper.getHeader().getStatus() == 0);
    }

    /**
     * 测试大文件的分片上传 超时
     * @throws Exception
     */
    @Test
    public void testUploadFileMultiWithTimeOut() throws Exception {
        UploadFileRequestWrapper uploadFileRequestWrapper = new UploadFileRequestWrapper();
        uploadFileRequestWrapper.setHeader(header);

        File file = new File("/tmp/b.txt");

        UploadFileRequest uploadFileRequest = new UploadFileRequest();
        uploadFileRequest.setFile(file);

        UploadFileParams uploadFileParams = new UploadFileParams();
        uploadFileParams.setFileName("test" + System.currentTimeMillis() + ".txt");
        uploadFileParams.setFileType("txt");
        uploadFileParams.setStoreType("temp");
        uploadFileParams.setBizFlag("audience");
        uploadFileRequest.setParams(uploadFileParams);
        uploadFileRequest.setTimeout(1000); // 设置1000ms

        uploadFileRequestWrapper.setBody(uploadFileRequest);

        UploadFileResponseWrapper uploadFileResponseWrapper = service.uploadFile(uploadFileRequestWrapper);
        System.out.println(uploadFileResponseWrapper);
        Assert.assertTrue(uploadFileResponseWrapper.getHeader().getStatus() == 0);
    }

    /**
     * 测试大文件上传 参数调优
     * @throws Exception
     */
    @Test
    public void testUploadFileMultiWithOptions() throws Exception {
        UploadFileRequestWrapper uploadFileRequestWrapper = new UploadFileRequestWrapper();
        uploadFileRequestWrapper.setHeader(header);

        File file = new File("/tmp/a.txt");

        UploadFileRequest uploadFileRequest = new UploadFileRequest();
        uploadFileRequest.setFile(file);

        UploadFileParams uploadFileParams = new UploadFileParams();
        uploadFileParams.setFileName("test" + System.currentTimeMillis() + ".txt");
        uploadFileParams.setFileType("txt");
        uploadFileParams.setStoreType("temp");
        uploadFileParams.setBizFlag("audience");
        uploadFileRequest.setParams(uploadFileParams);
        MultiUploadOptions options = new MultiUploadOptions();
        options.setMultipartThreshold(1024); // 1kb
        options.setPartSize(100 * 1024); // 100kb 最小100kb
        uploadFileRequest.setUploadOptions(options);

        uploadFileRequestWrapper.setBody(uploadFileRequest);

        UploadFileResponseWrapper uploadFileResponseWrapper = service.uploadFile(uploadFileRequestWrapper);
        System.out.println(uploadFileResponseWrapper);
        Assert.assertTrue(uploadFileResponseWrapper.getHeader().getStatus() == 0);
    }



    /**
     * 测试大文件上传 参数调优
     * @throws Exception
     */
    @Test
    public void testUploadFileMultiWithOptions2() throws Exception {
        UploadFileRequestWrapper uploadFileRequestWrapper = new UploadFileRequestWrapper();
        uploadFileRequestWrapper.setHeader(header);

        File file = new File("/tmp/b.txt");

        UploadFileRequest uploadFileRequest = new UploadFileRequest();
        uploadFileRequest.setFile(file);

        UploadFileParams uploadFileParams = new UploadFileParams();
        uploadFileParams.setFileName("test" + System.currentTimeMillis() + ".txt");
        uploadFileParams.setFileType("txt");
        uploadFileParams.setStoreType("temp");
        uploadFileParams.setBizFlag("audience");
        uploadFileRequest.setParams(uploadFileParams);
        MultiUploadOptions options = new MultiUploadOptions();
        options.setMultipartThreshold(100 * 1024 * 1024); // 100MB
        options.setPartSize(50 * 1024 * 1024); // 50MB
        uploadFileRequest.setUploadOptions(options);

        uploadFileRequestWrapper.setBody(uploadFileRequest);

        UploadFileResponseWrapper uploadFileResponseWrapper = service.uploadFile(uploadFileRequestWrapper);
        System.out.println(uploadFileResponseWrapper);
        Assert.assertTrue(uploadFileResponseWrapper.getHeader().getStatus() == 0);
    }
}
